### DISCUSSION TITLE
Should "west zephyr-export" be removed from the getting started guide?

### DISCUSSION THREAD

**Discussion by diegoherranz** [zephyrproject-rtos/zephyr#61039] [General] [answered]
The getting started guide (https://docs.zephyrproject.org/latest/develop/getting_started/index.html#get-zephyr-and-install-python-dependencies) shows a step to export a cmake package for Zephyr.

> Export a [Zephyr CMake package](https://docs.zephyrproject.org/latest/build/zephyr_cmake_package.html#cmake-pkg). This allows CMake to automatically load boilerplate code required for building Zephyr applications.
west zephyr-export

If I understand it correctly that may not be needed for the getting started guide since zephyr will be part of the workspace anyway and it may only be needed for freestanding applications.

Assuming that's correct (please let me know otherwise), would it make sense to remove that step from the getting started guide to make it simpler for beginners and avoid confusing them with cmake stuff? Or is it really needed for anything else?

Just let me know and if it makes sense, I could send a PR to modify the documentation.

Thanks!

**Comment by nashif** [comment]
@tejlmand  FYI

**Comment by tejlmand** [comment, marked as answer]
it is needed, even for getting started.

But if using `west`, then `west` will set `ZEPHYR_BASE` before invoking CMake, and then build will work, but remember that you may also use plain CMake, in which case `ZEPHYR_BASE` is not set in environment and is therefore found from the CMake package registry.
